src/bootstrap/lib.rs

```diff
@@ -104,7 +104,7 @@
 //! also check out the `src/bootstrap/README.md` file for more information.
 
 // NO-RUSTC-WRAPPER
-#![deny(warnings, rust_2018_idioms, unused_lifetimes)]
+#![deny(rust_2018_idioms, unused_lifetimes)]
```

Comment:
There are several crates with `NO-RUSTC-WRAPPER`.
Supposedly all of them should use this logic + `deny-warnings` should affect the rust_2018_idioms/unused_lifetimes stuff too.

I'm not sure how much code is affected by `RUSTFLAGS` set in `bootstrap.py`.
Perhaps the `#![deny(warnings, rust_2018_idioms, unused_lifetimes)]` can be moved to there?
Or perhaps `bootstrap.py` can define some feature and `NO-RUSTC-WRAPPER` can then use `#![cfg_attr(featuer_name, deny(warnings, rust_2018_idioms, unused_lifetimes))]`?

Reply:
(Warning levels for code going through the wrapper are set in https://github.com/rust-lang/rust/blob/master/src/bootstrap/bin/rustc.rs#L125-L136.)

Reply:
Well, bootstrap is special as it's the only crate built from bootstrap.py and not from rustbuild.
The environment modified in this PR is only used for this crate and not for the other ones

Reply:
Done